admin/module/catalog-admin-module-sources/src/main/webapp/js/view/ModalSource.view.js

```diff
@@ -162,7 +162,10 @@ function (ich,Marionette,Backbone,ConfigurationEdit,Service,Utils,wreqr,_,$,moda
                 model.save().then(function() {
                     var needsRefresh = true;
                     var existingSource = view.parentModel.get('collection').find(function(item) {
-                        return item.get('name') === view.getId(model);
+                        return (item &&
+                                item.attributes &&
+                                item.attributes.currentConfiguration &&
+                                item.attributes.currentConfiguration.id === view.getPID(model));
```

Comment:
use item.get('currentConfiguration') here instead of accessing attributes directly. Attributes is one of the backbone internal structures so it could change with an update to backbone.